lib/librte_hash/rte_hash.h

```diff
@@ -61,6 +61,11 @@ struct rte_hash_parameters {
 /** @internal A hash table structure. */
 struct rte_hash;
 
+/** @internal A hash table iterator state structure. */
+struct rte_iterator_state {
```

Comment:
`struct rte_iterator_state` should be named `struct rte_hash_iterator_state` to make it clear that it doesn't apply to the LPM table library for example.

To avoid a name conflic with the current `struct rte_hash_iterator_state`, rename `struct rte_hash_iterator_state` to `struct rte_hash_iterator_istate`. To keep the code uniform, rename `struct rte_hash_iterator_conflict_entries_state` to `struct rte_hash_iterator_conflict_entries_istate` as well.

Add the following comment before the definition of `struct rte_hash_iterator_istate` to help readers understand what the `i` in `istate` mean:
`/* istate stands for internal state. */`